ethers/provider.nim

```diff
@@ -227,21 +226,28 @@ proc confirm*(
   tx: TransactionResponse,
   confirmations = EthersDefaultConfirmations,
   timeout = EthersReceiptTimeoutBlks): Future[TransactionReceipt]
-  {.async: (raises: [CancelledError, ProviderError, EthersError]).} =
+  {.async: (raises: [CancelledError, ProviderError, SubscriptionError, EthersError]).} =
 
   ## Waits for a transaction to be mined and for the specified number of blocks
   ## to pass since it was mined (confirmations).
   ## A timeout, in blocks, can be specified that will raise an error if too many
   ## blocks have passed without the tx having been mined.
 
   var blockNumber: UInt256
+  var blockSubscriptionError: ref SubscriptionError
   let blockEvent = newAsyncEvent()
 
   proc onBlockNumber(number: UInt256) =
     blockNumber = number
     blockEvent.fire()
 
-  proc onBlock(blck: Block) =
+  proc onBlock(blckResult: SubscriptionResult[Block]) =
+    if blckResult.isErr:
+      blockSubscriptionError = blckResult.error()
+      blockEvent.fire()
+
+    let blck = blckResult.value
```

Comment:
Needs a `return` in the `if` block. Also, `blockSubscriptionError = nil` should be set somewhere to "reset" the error?

Can also do:
```nim
blockSubscriptionError = nil

without blck =? blckResult, error:
  blockSubscriptionError = error
  blockEvent.fire()
```